modules/reindex/src/main/java/org/elasticsearch/index/reindex/TransportReindexAction.java

```diff
@@ -194,16 +196,25 @@ protected ScrollableHitSource buildScrollableResultSource(BackoffPolicy backoffP
                 RestClientBuilder restClient = RestClient
                         .builder(new HttpHost(remoteInfo.getHost(), remoteInfo.getPort(), remoteInfo.getScheme()))
                         .setDefaultHeaders(clientHeaders);
-                if (remoteInfo.getUsername() != null) {
-                    restClient.setHttpClientConfigCallback(c -> {
+                restClient.setHttpClientConfigCallback(c -> {
+                    // Enable basic auth if it is configured
+                    if (remoteInfo.getUsername() != null) {
                         UsernamePasswordCredentials creds = new UsernamePasswordCredentials(remoteInfo.getUsername(),
                                 remoteInfo.getPassword());
                         CredentialsProvider credentialsProvider = new BasicCredentialsProvider();
                         credentialsProvider.setCredentials(AuthScope.ANY, creds);
                         c.setDefaultCredentialsProvider(credentialsProvider);
-                        return c;
+                    }
+                    // Stick the task id in the thread name so we can track down tasks from stack traces
+                    AtomicInteger threads = new AtomicInteger();
+                    c.setThreadFactory(r -> {
+                       String name = "es-client-" + task.getId() + "-" + threads.getAndIncrement();
+                       return new Thread(r, name);
                     });
-                }
+                    // Limit ourselves to one reactor thread because for now the search process is single threaded.
+                    c.setDefaultIOReactorConfig(IOReactorConfig.custom().setIoThreadCount(1).build());
```

Comment:
not too sure, but do we also want to test that these changes take effect somehow?